src/Microsoft.DotNet.TemplateLocator/TemplateLocator.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using NuGet.Common;
+using NuGet.Protocol;
+
+namespace Microsoft.DotNet.TemplateLocator
+{
+    public sealed class TemplateLocator
+    {
+        private DirectoryInfo _dotnetSdkTemplatesLocation;
+
+        public TemplateLocator()
+        {
+            string mockMockTemplateLocation = Environment.GetEnvironmentVariable("MOCKDOTNETSDKTEMPLATESLOCATION");
+            if (!string.IsNullOrWhiteSpace(mockMockTemplateLocation))
+            {
+                _dotnetSdkTemplatesLocation = new DirectoryInfo(mockMockTemplateLocation);
+            }
+        }
+
+        public IReadOnlyCollection<IOptionalSdkTemplatePackageInfo> GetDotnetSdkTemplatePackages(string sdkVersion)
+        {
+            if (_dotnetSdkTemplatesLocation == null)
+            {
+                return Array.Empty<IOptionalSdkTemplatePackageInfo>();
+            }
+
+            return LocalFolderUtility
+                .GetPackagesV2(_dotnetSdkTemplatesLocation.FullName, new NullLogger())
+                .Select(l => new OptionalSdkTemplatePackageInfo(l)).ToArray();
+        }
+
+        public string DotnetSdkVersionUsedInVs()
+        {
+            return "5.1.100";
+        }
+
+        internal void SetDotnetSdkTemplatesLocation(DirectoryInfo directoryInfo)
+        {
+            _dotnetSdkTemplatesLocation = directoryInfo;
+        }
+
+        private class OptionalSdkTemplatePackageInfo : IOptionalSdkTemplatePackageInfo
+        {
+            public OptionalSdkTemplatePackageInfo(LocalPackageInfo localPackageInfo)
```

Comment:
Can this ever be null?

Reply:
no. it is an internal class. The call site won't pass null